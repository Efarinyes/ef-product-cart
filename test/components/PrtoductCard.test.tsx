import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components';
import { product2 } from '../data/products';

const { act } = renderer;


describe('ProductCard', () => {
    test('Ha de mostrar el component de manera correcta', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <h1>Product Card </h1>
                    )
                }
            </ProductCard>
        )
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('El comptador ha de funcionar correctament', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    ({count, incressBy}) => (
                        <>
                            <h1>Product Card </h1>
                            <span>{ count }</span>
                            <button onClick={ () => incressBy(1)}></button>
                        </>
                    )
                }
            </ProductCard>
        );
        let tree = wrapper.toJSON();
        expect(tree).toMatchSnapshot();

        act(() => {
            (tree as any ).children[2].props.onClick();
        })
        tree = wrapper.toJSON();
        expect(( tree as any).children[1].children[0]).toBe('1');
       
    })
})

