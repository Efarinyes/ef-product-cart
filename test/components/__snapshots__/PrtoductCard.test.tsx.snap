// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard El comptador ha de funcionar correctament 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card 
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`ProductCard Ha de mostrar el component de manera correcta 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card 
  </h1>
</div>
`;
